File: leetcode/sorting/twoSum2.py
def twoSum(self, numbers: List[int], target: int) -> List[int]:
    d = {}
    for i in range(len(numbers)):
        t = target - numbers[i]
        if t in d:
            return [1 + d[t], 1 + i]
        d[numbers[i]] = i

File: leetcode/sorting/twoSum2.cpp
vector<int> twoSum(vector<int>& nums, int target) {
	vector<int> res;
	int i = 0; 
	int j = nums.size()-1;
	int sum = 0;
	while(i < j) {
		sum = nums[i] + nums[j];
		if (sum == target) {
			return {i+1,j+1};
		}
		if(sum > target) {
			j--;
		}
		else {
			i++;
		}
	}
	// empty vector
	return {};
}
